src/sentry/snuba/discover.py

```diff
@@ -457,11 +458,21 @@ def get_pagination_ids(event, query, params, reference_event=None, referrer=None
         if ref_conditions:
             snuba_filter.conditions.extend(ref_conditions)
 
+    def into_pagination_record(project_slug_event_id):
+
+        if project_slug_event_id is None:
+            return PaginationRecord(None, None)
+
+        project = Project.objects.get(
+            id=project_slug_event_id[0], organization=organization, status=ProjectStatus.VISIBLE
+        )
```

Comment:
Could we batch the project queries? Often this will do 4 queries for the same value.

Reply:
What's the preferred way to do this? Is it `in_bulk`? https://docs.djangoproject.com/en/1.11/ref/models/querysets/#in-bulk

Reply:
I would use `Project.objects.filter(id__in=projectors, ...)`